gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/FlowLaunchHandler.java

```diff
@@ -254,25 +256,27 @@ public static Properties getJobPropertiesFromJobDetail(JobDetail jobDetail) {
   @VisibleForTesting
   public static JobDataMap updatePropsInJobDataMap(JobDataMap jobDataMap,
       LeaseAttemptStatus.LeasedToAnotherStatus leasedToAnotherStatus, int schedulerMaxBackoffMillis) {
-    Properties prevJobProps = (Properties) jobDataMap.get(GobblinServiceJobScheduler.PROPERTIES_KEY);
+    JobDataMap newJobDataMap = (JobDataMap) jobDataMap.clone();
```

Comment:
This was already done when you did `JobDetailImpl jobDetail = (JobDetailImpl) this.schedulerService.getScheduler().getJobDetail(originalKey).clone();`
so it is a cloning of a cloned datamap

Reply:
No the cloning above makes a new reference to `jobDetail` but it's a shallow copy so without doing that clone the assertions of `notSame` fail (tested)

Reply:
that's strange because `JobDetailImpl.clone()` does
`copy.jobDataMap = (JobDataMap) jobDataMap.clone();`
which is same as this line.